### PR TITLE
bug(settings,content): Make banner x aligned right

### PR DESCRIPTION
**Pull request by dschom**
## Because

- We want the banner close icon to be aligned right

## This pull request

- Aligns the X icon by using mr-auto (ml-auto for rtl) for flex

## Issue that this pull request solves

Closes: FXA-8455

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)
Backbone:
image
image
image
image

React:
image
image
image
image
image
image

## Other information (Optional)

Note, that when testing the React version in Hebrew, the text direction wasn't swapping properly, where as for backbone it was. To get around this, I manually changed the `dir` attribute on the document, and you can see that once this is changed the layout renders correctly. I will be filing a follow up issue for the fact that Hebrew is not triggering a RTL direction in our react app. This defect is also observable on production.

### CODE REVIEW

File: packages/fxa-content-server/app/scripts/templates/partial/brand-messaging.mustache [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <div class="banner-brand-message w-full">
         <div class="flex justify-center p-2 brand-banner-bg">
             {{#showPrelaunch}}
-            <div class="flex">
+            <div class="flex ltr:ml-auto rtl:mr-auto">
```
Comment (vpomerleau) [inline packages/fxa-content-server/app/scripts/templates/partial/brand-messaging.mustache:6]:
```suggestion
            <div class="flex ms-auto">
```

File: packages/fxa-content-server/app/scripts/templates/partial/brand-messaging.mustache [hunk 2]
```diff
@@ -32,7 +32,7 @@
                 </div>
             </div>
             {{/showPostlaunch}}
-            <div class="flex justify-right order-last rtl:justify-left m-4 mt-1 mb-1">
+            <div class="flex justify-right order-last m-4 mt-1 mb-1 ml-auto rtl:justify-left rtl:ml-4 rtl:mr-auto">
```
Comment (vpomerleau) [inline packages/fxa-content-server/app/scripts/templates/partial/brand-messaging.mustache:35]:
```suggestion
            <div class="flex justify-right order-last m-4 mt-1 mb-1 ms-auto">
```

File: packages/fxa-settings/src/components/BrandMessaging/index.tsx [hunk 3]
```diff
@@ -98,7 +98,10 @@ export const BrandMessaging = ({
     >
       <div className="flex relative justify-center p-2 brand-banner-bg border border-transparent">
         {mode === 'prelaunch' && (
-          <div className="flex" data-testid="brand-prelaunch">
+          <div
+            className="flex ltr:ml-auto rtl:mr-auto"
```
Comment (vpomerleau) [inline packages/fxa-settings/src/components/BrandMessaging/index.tsx:102]:
```suggestion
            className="flex ms-auto"
```

File: packages/fxa-settings/src/components/BrandMessaging/index.tsx [hunk 4]
```diff
@@ -134,7 +137,10 @@ export const BrandMessaging = ({
         )}
 
         {mode === 'postlaunch' && (
-          <div className="flex" data-testid="brand-postlaunch">
+          <div
+            className="flex ltr:ml-auto rtl:mr-auto"
```
Comment (vpomerleau) [inline packages/fxa-settings/src/components/BrandMessaging/index.tsx:141]:
```suggestion
            className="flex ms-auto"
```

**Review by vpomerleau — CHANGES_REQUESTED** [review]
We could instead use logical properties (ms-auto), it's a bit easier to read: https://tailwindcss.com/docs/margin#using-logical-properties

**Comment by dschom** [comment]
> We could instead use logical properties (ms-auto), it's a bit easier to read: https://tailwindcss.com/docs/margin#using-logical-properties

@vpomerleau much cleaner. I switched to this, retested, and pushed updates.

**Review by vpomerleau — APPROVED** [review]
Sweet! Thanks for the update! :shipit:

**Pull request merged**